vmdb/app/models/host_vmware_esx.rb

```diff
@@ -106,22 +116,24 @@ def vim_advanced_settings
     EmsRefresh::Parsers::Vc.host_inv_to_advanced_settings_hashes(data)
   end
 
-  def verify_credentials_with_ws(auth_type=nil)
+  def verify_credentials_with_ws(auth_type = nil)
     raise "No credentials defined" if self.missing_credentials?(auth_type)
 
     begin
       with_provider_connection(:use_broker => false, :auth_type => auth_type) {}
     rescue SocketError, Errno::EHOSTUNREACH, Errno::ENETUNREACH
-      raise MiqException::MiqUnreachableError, $!.message
+      raise MiqException::MiqUnreachableError, $ERROR_INFO.message
```

Comment:
Even more preferable is to not use the $ variable at all.  You can rescue the exception into a variable and use that.  Typically people use the variable `err`

```ruby
rescue SocketError, Errno::EHOSTUNREACH, Errno::ENETUNREACH => err
  raise MiqException::MiqUnreachableError, err.message
```